From 7891776c9d1aa7c1499b91efffe145183f12625a Mon Sep 17 00:00:00 2001
From: Yuta Saito <kateinoigakukun@gmail.com>
Date: Thu, 7 Mar 2024 18:27:46 +0000
Subject: [PATCH] [wasm] Port CoreFoundation/Locale.subproj

* Remove no longer needed `#if !TARGET_OS_WASI`
* Add WASI to the list of ICU supported platforms
---
 CoreFoundation/Locale.subproj/CFLocale.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/CoreFoundation/Locale.subproj/CFLocale.c b/CoreFoundation/Locale.subproj/CFLocale.c
index 738ff28a..2c2a942d 100644
--- a/CoreFoundation/Locale.subproj/CFLocale.c
+++ b/CoreFoundation/Locale.subproj/CFLocale.c
@@ -19,15 +19,11 @@
 #include <CoreFoundation/CFNumber.h>
 #include "CFInternal.h"
 #include "CFRuntime_Internal.h"
-#if !TARGET_OS_WASI
 #include <CoreFoundation/CFPreferences.h>
 #include "CFBundle_Internal.h"
-#else
-#include "CFBase.h"
-#endif
 #include "CFLocaleInternal.h"
 #include <stdatomic.h>
-#if TARGET_OS_MAC || TARGET_OS_WIN32 || TARGET_OS_LINUX || TARGET_OS_BSD
+#if TARGET_OS_MAC || TARGET_OS_WIN32 || TARGET_OS_LINUX || TARGET_OS_BSD || TARGET_OS_WASI
 #include <unicode/uloc.h>           // ICU locales
 #include <unicode/ulocdata.h>       // ICU locale data
 #include <unicode/ucal.h>
-- 
2.43.0

